posixlib/src/main/scala/scala/scalanative/posix/signal.scala

```diff
@@ -0,0 +1,411 @@
+package scala.scalanative
+package posix
+
+import scala.scalanative.unsafe._
+
+/**
+ * Some of the functionality described on this reference page extends the ISO C standard.
+ * Applications shall define the appropriate feature test macro (see XSH The Compilation Environment)
+ * to enable the visibility of these symbols in this header.
+ *
+ * Use clib macros and functions to get the full POSIX interface
+ *
+ * Note 1: The functionality described may be removed in a future version of this volume of POSIX.1-2017
+ * Note 2: The functionality is an option marked as XSI (X/Open System Interfaces)
+ *
+ * Note 1 or 2: followed by Linux or macOS means that the feature is not available on that platform with
+ * default compile options so the code is commented out. The code is commented out so the next developer
+ * that comes along will not have to rediscover these limitations.
+ */
+@extern
+object signal {
+  // define the following macros, which shall expand to constant expressions with distinct values
+  // that have a type compatible with the second argument to, and the return value of, the signal() function,
+  // and whose values shall compare unequal to the address of any declarable function.
+
+  // Note 1: Linux
+  // @name("scalanative_sig_hold")
+  // def SIG_HOLD: CFunctionPtr1[CInt, Unit] = extern
+
+  import sys.types
+  // pthread_t, size_t, and uid_t types as described in <sys/types.h>
+  type pthread_t = types.pthread_t
+  type size_t    = types.size_t
+  type uid_t     = types.uid_t
+
+  // timespec structure as described in <time.h>
+  type timespec = time.timespec
+
+  // define the following data types
+  type sig_atomic_t = CInt
```

Comment:
Granted all the discussions about SN, uni-threading, and multithreading.
Granted also that `concurrent/atomic` belies it name in the multighread
case.

Since this is posixlib, please consider using `javalib/src/main/scala/java/util/concurrent/atomic/AtomicInteger.scala`.

Reply:
They are all just `int`s so I am guess this is correct or possible generated.
>sig_atomic_t
    Possibly volatile-qualified integer type of an object that can be accessed as an atomic entity, even in the presence of asynchronous interrupts.